raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientSnapshotTest.java

```diff
@@ -0,0 +1,765 @@
+            assertEquals(0, response.position());
+            assertEquals(snapshot.sizeInBytes() / 2, response.bytes().remaining());
+
+            ByteBuffer snapshotBuffer = ByteBuffer.allocate(Math.toIntExact(snapshot.sizeInBytes()));
+            snapshot.read(snapshotBuffer, 0);
+            snapshotBuffer.flip();
+
+            ByteBuffer responseBuffer = ByteBuffer.allocate(Math.toIntExact(snapshot.sizeInBytes()));
+            responseBuffer.put(response.bytes());
+
+            ByteBuffer expectedBytes = snapshotBuffer.duplicate();
+            expectedBytes.limit(Math.toIntExact(snapshot.sizeInBytes() / 2));
+
+            assertEquals(expectedBytes, responseBuffer.duplicate().flip());
+
+            // Fetch the remainder of the snapshot
+            context.deliverRequest(
+                fetchSnapshotRequest(
+                    context.metadataPartition,
+                    epoch,
+                    snapshotId,
+                    Integer.MAX_VALUE,
+                    responseBuffer.position()
+                )
+            );
+
+            context.client.poll();
+
+            response = context.assertSentFetchSnapshotResponse(context.metadataPartition).get();
+            assertEquals(Errors.NONE, Errors.forCode(response.errorCode()));
+            assertEquals(snapshot.sizeInBytes(), response.size());
+            assertEquals(responseBuffer.position(), response.position());
+            assertEquals(snapshot.sizeInBytes() - (snapshot.sizeInBytes() / 2), response.bytes().remaining());
+
+            responseBuffer.put(response.bytes());
+            assertEquals(snapshotBuffer, responseBuffer.flip());
+        }
+    }
+
+    @Test
+    public void testFetchSnapshotRequestAsFollower() throws IOException {
+        int localId = 0;
+        int leaderId = localId + 1;
+        Set<Integer> voters = Utils.mkSet(localId, leaderId);
+        int epoch = 2;
+        OffsetAndEpoch snapshotId = new OffsetAndEpoch(0, 0);
+
+        RaftClientTestContext context = new RaftClientTestContext.Builder(localId, voters)
+            .withElectedLeader(epoch, leaderId)
+            .build();
+
+        context.deliverRequest(
+            fetchSnapshotRequest(
+                context.metadataPartition,
+                epoch,
+                snapshotId,
+                Integer.MAX_VALUE,
+                0
+            )
+        );
+
+        context.client.poll();
+
+        FetchSnapshotResponseData.PartitionSnapshot response =  context.assertSentFetchSnapshotResponse(context.metadataPartition).get();
+        assertEquals(Errors.NOT_LEADER_OR_FOLLOWER, Errors.forCode(response.errorCode()));
+        assertEquals(epoch, response.currentLeader().leaderEpoch());
+        assertEquals(leaderId, response.currentLeader().leaderId());
+    }
+
+    @Disabled
+    @Test
+    public void testFetchSnapshotRequestWithOlderEpoch() throws IOException {
+        assertTrue(false);
```

Comment:
I guess you're still planning to implement these?

Reply:
Yes. Doing that now...